bhmm/tests/benchmark_hidden.py

```diff
@@ -0,0 +1,180 @@
+        self.pi.append(pi)
+        self.pobs.append(pobs)
+        self.N.append(A.shape[0])
+        self.T.append(pobs.shape[0])
+        # compute intermediates
+        _, alpha, _ = self.run_forward(i, None)
+        self.alpha.append(alpha)
+        beta, _ = self.run_backward(i, None)
+        self.beta.append(beta)
+        gamma, _ = self.run_gamma(i, None)
+        self.gamma.append(gamma)
+        #
+        self.alpha_mem.append(np.zeros((pobs.shape[0],A.shape[0])))
+        self.beta_mem.append(np.zeros((pobs.shape[0],A.shape[0])))
+        self.gamma_mem.append(np.zeros((pobs.shape[0],A.shape[0])))
+        self.C_mem.append(np.zeros((A.shape[0],A.shape[0])))
+
+        self.nexamples += 1
+
+    def run_forward(self, i, out):
+        logprob = 0
+        alpha = None
+        hidden.set_implementation(self.kernel)
+        time1 = time.time()
+        for k in range(self.nrep):
+            logprob, alpha = hidden.forward(self.A[i], self.pobs[i], self.pi[i], alpha_out=out)
+        # compare
+        time2 = time.time()
+        d = (time2-time1)/(1.0*self.nrep)
+        return (logprob, alpha, d)
+
+    def run_backward(self, i, out):
+        beta = None
+        hidden.set_implementation(self.kernel)
+        time1 = time.time()
+        for k in range(self.nrep):
+            beta = hidden.backward(self.A[i], self.pobs[i], beta_out=out)
+        # compare
+        time2 = time.time()
+        d = (time2-time1)/(1.0*self.nrep)
+        return (beta, d)
+
+    def run_gamma(self, i, out):
+        gamma = None
+        hidden.set_implementation(self.kernel)
+        time1 = time.time()
+        for k in range(self.nrep):
+            gamma = hidden.state_probabilities(self.alpha[i], self.beta[i], gamma_out=out)
+        # compare
+        time2 = time.time()
+        d = (time2-time1)/(1.0*self.nrep)
+        return (gamma, d)
+
+    def run_state_counts(self, i, out):
+        c = None
+        hidden.set_implementation(self.kernel)
+        time1 = time.time()
+        for k in range(self.nrep):
+            c = hidden.state_counts(self.gamma[i], self.T[i])
+        # compare
+        time2 = time.time()
+        d = (time2-time1)/(1.0*self.nrep)
+        return (c, d)
+
+    def run_transition_counts(self, i, out):
+        C = None
+        hidden.set_implementation(self.kernel)
+        time1 = time.time()
+        for k in range(self.nrep):
+            C = hidden.transition_counts(self.alpha[i], self.beta[i], self.A[i], self.pobs[i], out=out)
+        # compare
+        time2 = time.time()
+        d = (time2-time1)/(1.0*self.nrep)
+        return (C, d)
+
+    def run_viterbi(self, i, out):
+        vpath = None
+        hidden.set_implementation(self.kernel)
+        time1 = time.time()
+        for k in range(self.nrep):
+            vpath = hidden.viterbi(self.A[i], self.pobs[i], self.pi[i])
+        # compare
+        time2 = time.time()
+        d = (time2-time1)/(1.0*self.nrep)
+        return (vpath, d)
```

Comment:
These returns only return the *last* result.  Did you want to append all the replicates instead?